src/coreclr/jit/valuenum.cpp

```diff
@@ -9965,6 +9965,19 @@ void Compiler::fgValueNumberSsaVarDef(GenTreeLclVarCommon* lcl)
 //
 static bool GetStaticFieldSeqAndAddress(ValueNumStore* vnStore, GenTree* tree, ssize_t* byteOffset, FieldSeq** pFseq)
 {
+    VNFuncApp funcApp;
+    if (tree->TypeIs(TYP_REF) && vnStore->GetVNFunc(tree->gtVNPair.GetLiberal(), &funcApp) &&
```

Comment:
Shouldn't this be `TYP_BYREF`?

Reply:
Is there any point in checking the type of the address?

Reply:
> Is there any point in checking the type of the address?

no, removed 🙂